### PR TITLE
ODPM-80 MD and NC API unit tests

### PR DESCRIPTION
**Pull request by trawick**
This takes the existing branch for NC API tests, gets it in sync with the current code, updates some of the tests to check more details, implements a consistent set of tests for MD, then adds a test of contraband-hit-rate for both.

The NC use-of-force endpoint (not relevant for MD) is not tested.

### CODE REVIEW

File: md/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from md.models import ETHNICITY_CHOICES, PURPOSE_CHOICES
+from md.tests import factories
+
+
+class AgencyTests(APITestCase):
+
+    # Why "aaa"?  It must run first to get the expected agency list.
```
Comment (nmashton) [inline md/tests/test_api.py]: Why does this happen?

Comment (trawick) [inline md/tests/test_api.py]: I think that if this testcase doesn't run first, there will be more than one agency in the list, and the testcase expects exactly one.  I'll try messing with it to make it more resilient.

File: nc/tests/test_api.py [hunk 2]
```diff
@@ -1,30 +1,143 @@
...
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 0)
+
+    def test_stops_count(self):
+        """Test Agency stop counts"""
+        agency = factories.AgencyFactory()
+        factories.PersonFactory(race='B', stop__agency=agency,
+                                ethnicity='N', stop__year=2010)
+        factories.PersonFactory(race='B', stop__agency=agency,
+                                ethnicity='N', stop__year=2010)
```
Comment (rlconley) [inline nc/tests/test_api.py]: Why are these two lines repeated?

**Comment by nmashton** [comment]
With the proviso that this is problematic and should be addressed (see below), 🚢 

image

File: md/tests/test_api.py (hunk 1 above)
Comment (trawick) [inline md/tests/test_api.py]: Addressed in ce78835

**Comment by trawick** [comment]
I almost forgot; the "..re to remain at scene of accident" truncation was created with ODPM-76 when trying to fix that unrelated bug.  Go karma!  I'll open a bug to track that separately.

**Comment by nmashton** [comment]
💵

**Pull request merged**